### PR TITLE
Align configuration with core

### PR DESCRIPTION
**Pull request by Necroneco**
fix BLE001 RET505 I001 and some docstrings

### CODE REVIEW

**Comment by Necroneco** [comment]
maybe should update `ruff.toml` first

**Comment by wuwentao** [comment]
@Necroneco 
just follow the error msg and update the file with the lint error log about the diff :
```
All changes made by hooks:
diff --git a/custom_components/midea_ac_lan/const.py b/custom_components/midea_ac_lan/const.py
index 872be60..a3bae4f 100644
--- a/custom_components/midea_ac_lan/const.py
+++ b/custom_components/midea_ac_lan/const.py
@@ -11,5 +11,12 @@ CONF_SERVER = "server"
 CONF_REFRESH_INTERVAL = "refresh_interval"
 EXTRA_SENSOR = [Platform.SENSOR, Platform.BINARY_SENSOR]
 EXTRA_SWITCH = [Platform.SWITCH, Platform.LOCK, Platform.SELECT, Platform.NUMBER]
-EXTRA_CONTROL = [Platform.CLIMATE, Platform.WATER_HEATER, Platform.FAN, Platform.HUMIDIFIER, Platform.LIGHT, *EXTRA_SWITCH]
+EXTRA_CONTROL = [
+    Platform.CLIMATE,
+    Platform.WATER_HEATER,
+    Platform.FAN,
+    Platform.HUMIDIFIER,
+    Platform.LIGHT,
+    *EXTRA_SWITCH,
+]
 ALL_PLATFORM = EXTRA_SENSOR + EXTRA_CONTROL
```

**Comment by wuwentao** [comment]
we need to use devcontainer and clone it in repo.
I'm also never use devcontainer, as we always using a dev server and works in it.
for github open source project, we may need to follow this mode.

for windows PC: install WSL2 and docker, and use vscode devcontainer extention 
https://code.visualstudio.com/blogs/2020/07/01/containers-wsl

https://learn.microsoft.com/zh-cn/windows/wsl/install
https://docs.docker.com/desktop/wsl/#download

as I‘m using MacOS and also have a windows PC.
just finished install WSL2 and docker  today.

**Comment by wuwentao** [comment]
manual install ruff and run ruff check/format with the error file, same as black/isort
https://github.com/astral-sh/ruff

**Comment by chemelli74** [comment]
> manual install ruff and run ruff check/format with the error file, same as black/isort https://github.com/astral-sh/ruff

We should install ruff via devcontainer if not yet.

**Comment by Necroneco** [comment]
Only the docstrings rules (`D`) have not been processed yet, temporarily disabled.

File: custom_components/midea_ac_lan/config_flow.py [hunk 1]
```diff
@@ -306,74 +302,72 @@ async def async_step_auto(
...
+                self.found_device[CONF_SUBTYPE] = device_info.get("model_number")
+            # get token and key from cloud for v3 device
+            if device.get(CONF_PROTOCOL) == 3:
+                if self.account[CONF_SERVER] == "美的美居":
+                    _LOGGER.debug(
+                        "Try to get the Token and the Key use the preset MSmartHome account",
+                    )
                     self.cloud = get_midea_cloud(
-                        self.account[CONF_SERVER],
+                        "MSmartHome",
```
Comment (chemelli74) [inline custom_components/midea_ac_lan/config_flow.py:337]: Can you explain why we need to hardcore a value here ?

File: custom_components/midea_ac_lan/fan.py [hunk 2]
```diff
@@ -105,15 +99,8 @@ async def async_set_percentage(self, percentage: int) -> None:
             await self.hass.async_add_executor_job(self.set_percentage, percentage)
 
     def update_state(self, status: Any) -> None:
-        try:
+        if self.hass:
```
Comment (chemelli74) [inline custom_components/midea_ac_lan/fan.py:102]: What's the purpose of the if ?

File: custom_components/midea_ac_lan/water_heater.py [hunk 3]
```diff
@@ -167,15 +165,8 @@ async def async_turn_off(self, **kwargs: Any) -> None:
         await self.hass.async_add_executor_job(ft.partial(self.turn_off, **kwargs))
 
     def update_state(self, status: Any) -> None:
-        try:
+        if self.hass:
```
Comment (chemelli74) [inline custom_components/midea_ac_lan/water_heater.py:168]: Here as well. Please explain.

File: pylint/ruff.toml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# This extend our general Ruff rules specifically for tests
```
Comment (chemelli74) [inline pylint/ruff.toml:1]: Is that temporary until we can enable the full ruff config ?

**Review by chemelli74 — CHANGES_REQUESTED** [review]

File: custom_components/midea_ac_lan/config_flow.py (hunk 1 above)
Comment (Necroneco) [reply custom_components/midea_ac_lan/config_flow.py:337]: This is a display bug of `git diff`, here only reduce the indentation

File: pylint/ruff.toml (hunk 4 above)
Comment (Necroneco) [reply pylint/ruff.toml:1]: I don't know, this is just copied from core

File: custom_components/midea_ac_lan/fan.py (hunk 2 above)
Comment (Necroneco) [reply custom_components/midea_ac_lan/fan.py:102]: By BLE001: Do not catch blind exception: `Exception`

As far as I know, the only exception to `self.schedule_update_ha_state` is when `self.hass` is `None`.
and here is just a debug log, I think it's okay to remove it.

Or just change `Exception` to `AttributeError` ?

Do you have any advice?

File: custom_components/midea_ac_lan/water_heater.py (hunk 3 above)
Comment (Necroneco) [reply custom_components/midea_ac_lan/water_heater.py:168]: Same as above

**Pull request closed without merging**